### PR TITLE
[CFE-497]: Feat(pages) - support migrate page with lcd

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/components/ContractInputSection.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  const [isChangeContract, setIsChangeContract] = useState(false);
+  const { address } = useCurrentChain();
+  const { contract: contractExample } = useExampleAddresses();
+  const { control, setValue, watch } = useForm({
+    defaultValues: {
+      contractAddress: "",
+    },
+  });
+  const { contractAddress } = watch();
+  const [debouncedKeyword, setDebouncedKeyword] = useState("");
```
Comment (evilpeach) [inline src/lib/components/ContractInputSection.tsx:35]: Please use `useDebounce`

File: src/lib/components/ContractInputSection.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  const { data, isFetching } = useContractData(debouncedKeyword as BechAddr32, {
+    enabled: !!debouncedKeyword,
+  });
+  const isPermissionAllowed = data?.contract.admin === address;
+
+  const handleInputStatus = (): FormStatus | undefined => {
```
Comment (evilpeach) [inline src/lib/components/ContractInputSection.tsx:42]: Apply useMemo

File: src/lib/components/ContractInputSection.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        state: "success",
+        message: "You have admin access to this contract",
+      };
+    return {
+      state: "error",
+      message: "Your wallet does not have admin access to this contract",
+    };
+  };
+
+  useEffect(() => {
```
Comment (evilpeach) [inline src/lib/components/ContractInputSection.tsx:60]: If you use `useDebounce`, it's no need to use it.

**Review by evilpeach — APPROVED** [review]

File: src/lib/pages/migrate/index.tsx [hunk 4]
```diff
@@ -193,7 +203,7 @@ const Migrate = () => {
         <FooterCTA
           cancelButton={{
             leftIcon: <CustomIcon name="chevron-left" />,
-            onClick: router.back,
+            onClick: () => handleBack(),
```
Comment (songwongtp) [inline src/lib/pages/migrate/index.tsx:206]: `onClick: handleBack`

**Review by songwongtp — APPROVED** [review]

**Pull request merged**